seminar1/발전과제/random.ts

```diff
@@ -0,0 +1,71 @@
+// 과제 조건
+// 1. Member, Dinner interface 만들고 타입 지정하기
+// 2. organize 내부 로직 채우기
+
+const OB = "ob";
+const YB = "yb";
+
+interface Member {
+  name: string;
+  group: string;
+}
+
+interface Dinner {
+  member: Array<Member>;
+  shuffle: Function;
+  organize: Function;
+}
+
+const dinner: Dinner = {
+  member: [
+    {
+      name: "채정아",
+      group: OB,
+    },
+    {
+      name: "김동재",
+      group: YB,
+    },
+    {
+      name: "강민재",
+      group: YB,
+    },
+    {
+      name: "김루희",
+      group: OB,
+    },
+    {
+      name: "박진수",
+      group: OB,
+    },
+  ],
+  shuffle(array: Array<Member>) {
+    array.sort(() => Math.random() - 0.5);
```

Comment:
`sort` 사용에 대해서 고민해보면 좋을 것 같습니다. `sort`는 브라우저 별로 implementation이 다릅니다. 그래서 동작 방식이 동일하다는 사실을 장담할 수 없습니다.

> Your second attempt, with sort(() => Math.random() - 0.5), returns with equal probability that the first number is greater than the second, or vice versa, which makes the shuffle work much better. However, because of the unreliability of the whole thing there's zero assurance that the shuffle will work in all browsers or be particularly random. Please use the "real" shuffle algorithm linked above.

[참고 1](https://stackoverflow.com/questions/2450954/how-to-randomize-shuffle-a-javascript-array)
[참고 2](https://262.ecma-international.org/6.0/#sec-array.prototype.sort)

Reply:
오호- 재밌는 사실이네요

Reply:
흥~미로운 사실이군요?